metagpt/ext/sela/runner/sela.py

```diff
@@ -0,0 +1,152 @@
+import argparse
+import json
+import os
+from typing import Optional
+
+from metagpt.ext.sela.runner.custom import CustomRunner
+from metagpt.ext.sela.runner.mcts import MCTSRunner
+from metagpt.ext.sela.runner.random_search import RandomSearchRunner
+from metagpt.ext.sela.runner.runner import Runner
+from metagpt.llm import LLM
+
+SELA_INSTRUCTION = """
+You are an assistant for configuring machine learning experiments.
+
+Given the requirement:
+{requirement}
+
+Your task:
+1. Extract **experiment configurations** from the requirement if explicitly mentioned, such as:
+   - "rollouts: 10"
+   - "exp_mode: mcts"
+   - "max_depth: 4"
+
+2. Extract **experiment data information**, including:
+   - **dataset**: Dataset name
+   - **metric**: Evaluation metric
+   - **target_col**: Target column
+   - **user_requirement**: Specific instructions or dataset handling requirements 
+
+Output a JSON object with two parts:
+- "config": A dictionary of explicitly mentioned configurations, using keys:
+  - "task": str (a noun based on the dataset name, customizable, e.g., "titanic")
+  - "exp_mode": str (e.g., "mcts", "rs", "base", "custom", "greedy", "autogluon")
+  - "rollouts": int
+  - "max_depth": int
+  - "rs_mode": str (e.g., "single", "set")
+  - "special_instruction": str (e.g., "text", "image")
+- "data_info": A dictionary of experiment data information, with keys:
+  - "dataset": str (e.g., "04_titanic")
+  - "metric": str (e.g., "f1", "rmse")
+  - "target_col": str (e.g., "Survived")
+  - "user_requirement": str 
+
+Example output:
+{{
+  "config": {{
+    "task": "titanic",
+    "exp_mode": "mcts",
+    "rollouts": 10
+  }},
+  "data_info": {{
+    "dataset": "04_titanic",
+    "metric": "f1",
+    "target_col": "Survived",
+    "user_requirement": "Predict the target column `Survived`. Perform data analysis, preprocessing, feature engineering, and modeling. Report f1 on eval data. Do not include visualizations."
+  }}
+}}
+
+Return only the JSON object.
+"""
+
+
+class Sela:
+    DEFAULT_CONFIG = {
+        "name": "",
+        "reflection": True,
+        "no_reflection": False,
+        "exp_mode": "mcts",
+        "rollouts": 10,
+        "load_tree": False,
+        "role_timeout": 1000,
+        "use_fixed_insights": False,
+        "low_is_better": False,
+        "start_task_id": 2,
+        "from_scratch": False,
```

Comment:
for the default config, maybe set `from_scratch` to be True since it's an end-to-end task and no insights are pre-generated